src/main/java/com/microsoft/azure/spring/data/documentdb/common/PropertyLoader.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.data.documentdb.common;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+public class PropertyLoader {
+    private static final String PROJECT_PROPERTY_FILE = "/META-INF/project.properties";
+
+    public static String getProjectVersion() {
+        String version = "unknown";
+        InputStream inputStream = null;
+        try {
+            inputStream = PropertyLoader.class.getResourceAsStream(PROJECT_PROPERTY_FILE);
+            if (inputStream != null) {
+                final Properties properties = new Properties();
+                properties.load(inputStream);
+
+                version = properties.getProperty("project.version");
+            }
+        } catch (IOException e) {
+            // Omitted
```

Comment:
So are we intentionally swallow the IOException here?

Reply:
Yes, basically this is a utility tool to retrieve the version, will ignore the IOException here and use default user agent directly.

Reply:
ok then.